src/Uno.UWP/UI/Notifications/ToastNotifier.Android.cs

```diff
@@ -0,0 +1,343 @@
+			{
+#pragma warning disable CS0618 // Type or member is obsolete
+				builder.SetPriority(0); // PRIORITY_DEFAULT
+#pragma warning restore CS0618 // Type or member is obsolete
+			}
+
+		}
+
+		private void SetToastTimeout(Android.App.Notification.Builder builder, DateTimeOffset? expirationTime)
+		{
+			if ((expirationTime != null) && (expirationTime.HasValue))
+			{
+				int seconds = (int)(expirationTime.Value - DateTime.Now).TotalSeconds;
+				if (seconds > 0)
+				{
+					builder.SetTimeoutAfter(seconds * 1000);
+				}
+			}
+		}
+
+		private Android.App.PendingIntent CreateToastIntent(string packageName, XmlDocument xmlDoc)
+		{
+			var intentAction = Android.App.Application.Context?.PackageManager?.GetLaunchIntentForPackage(packageName);
+			if (intentAction is null)
+			{
+				throw new InvalidOperationException("ToastNotifier:GetIconId, ResolveActivity is null");
+			}
+
+			intentAction.SetPackage(packageName);
+			// according to doc:
+			// The name must include a package prefix, for example the app com.android.contacts would use names like "com.android.contacts.ShowAll"
+			// but it works also in this way, and getting package name in background app can be difficult.
+			intentAction.PutExtra("Uno.internal.IntentType", (double)ApplicationModel.Activation.ActivationKind.ToastNotification);
+
+			string toastArgument = "";
+			var childToast = xmlDoc.GetElementsByTagName("toast");
+			if (childToast.Count == 1)
+			{
+				var childLaunch = childToast[0]?.Attributes?.GetNamedItem("launch");
+				if (childLaunch != null)
+				{
+					var launchValue = childLaunch.Value;
+					if (launchValue is null)
+					{
+						throw new Exception("It cannot happen, but to keep compiler happy...");
+					}
+					toastArgument = launchValue;
+				}
+			}
+
+			intentAction.PutExtra("Uno.internal.ToastArgument", toastArgument);
+
+			// we should have intent.data field different for different toast arguments,
+			// else - next intent would overwrite previous.
+			// it can be any URI, e.g. http://dummy.com/dummy?arg=  
+			var data = Android.Net.Uri.Parse("http://unotask.com/toast?arg=" + Android.Net.Uri.Encode(toastArgument));
+			intentAction.SetData(data);
+
+			// 12345: arbitrary number, it can be any value (as it is not used in Uno code anywhere)
+			var retVal = Android.App.PendingIntent.GetActivity(Android.App.Application.Context, 12345, intentAction, 0); // 12345=requestCode
+			if (retVal is null)
+			{
+				throw new InvalidOperationException("ToastNotifier:CreateToastIntent, GetActivity is null");
+			}
+
+			return retVal;
+		}
+
+
+		public void Show(ToastNotification notification)
+		{
+			var androidNotification = new Android.App.Notification();
+			androidNotification.Category = Android.App.Notification.CategoryMessage;
+
+			Android.App.Notification.Builder builder = GetToastBuilder();
+
+			// we use XmlDocument for retrieving text items and launch attribute

```

Comment:
```suggestion
			// we use XmlDocument for retrieving text items and launching attribute

```

Reply:
No, it is launch atrribute:
childToast[0]?.Attributes?.GetNamedItem("launch");